cobc/cobc.c

```diff
@@ -3341,6 +3343,7 @@ process_command_line (const int argc, char **argv)
 
 		case 'F':
 			/* --free, alias of `-fformat=free` */
+			cobc_free_format = 1 ;
 			(void) cobc_deciph_source_format ("FREE");
```

Comment:
Let's move the check function to pplex.l, this way we cover `-fformat=free`/`variable`..., too.
If not done already the default should be changed from "fixed" to "unknown", so that the check is only done if there wasn't no explicit format set.

Reply:
When adjusting `enum cb_format` to include `CB_FORMAT_UNKNOWN` as first entry (no need for the zero as we don't "serialize" that as int somewhere) then no `cobc_autodetect_source_format` is needed.

Reply:
ping @lefessan: Using a default of `source_format = CB_FORMAT_UNKNOWN;` we neither need `cobc_autodetect_source_format` nor a new `main_source` parameter, do we?

Reply:
I set it to `CB_FORMAT_AUTO` by default and got rid of `cobc_autodetect_source_format` to get the same behavior.